internal/discoverer/zookeeper.go

```diff
@@ -47,37 +48,36 @@ func (zd *ZookeeperDiscoverer) Stop() {
 	zd.zkUnWatchCancel()
 }
 
-func (zd *ZookeeperDiscoverer) Query(query *comm.Query) error {
-	values, _, err := query.Decode()
-	if err != nil {
-		return err
-	}
-
-	event, entity, serviceName := values[0], values[1], values[2]
-
-	switch event {
-	case utils.EventAdd:
-		err = zd.fetchService(serviceName, []string{entity})
-	case utils.EventDelete:
-		err = zd.removeService(serviceName)
+func (zd *ZookeeperDiscoverer) Query(msg *message.Message) error {
+	var err error
+	switch msg.Action {
+	case message.EventAdd:
+		err = zd.fetchService(msg.ServiceName(), map[string]*message.Message{msg.Key: msg})
+	case message.EventDelete:
+		err = zd.removeService(msg.ServiceName())
 	}
 
 	return err
 }
 
-func (zd *ZookeeperDiscoverer) Update(update *comm.Update) error {
+func (zd *ZookeeperDiscoverer) Update(oldMsg, msg *message.Message) error {
 	// TODO:
 	// Zookeeper does not need to introduce parameter information like nacos,
 	// so the Update interface cannot be triggered on the framework.
 	return nil
 }
 
-func (zd *ZookeeperDiscoverer) Watch() chan *comm.Message {
+func (zd *ZookeeperDiscoverer) Delete(msg *message.Message) error {
+	// TODO:
+	return nil
```

Comment:
We need to handle it in `discoverer.GetDiscoverer(delMsg.DiscoveryType()).Delete(delMsg)`?

Reply:
Yes, it is needed. And cc@shuaijinchao ,can you assist me in implementing this function?